test/Spec/MLabsPAB/Contract.hs

```diff
@@ -29,108 +49,278 @@ tests =
+        let constraints =
+              Constraints.mustPayToPubKey
+                pkh2
+                (Ada.lovelaceValueOf 1000 <> Value.singleton "abcd1234" "testToken" 5)
+        void $ submitTx constraints
+
+      (result, state, _) = runContractPure contract initState
+
+  result @?= Right ()
+  (state.commandHistory !! 1)
+    @?= Text.replace
+      "\n"
+      " "
+      [text|
+        cardano-cli transaction build --alonzo-era
+        --tx-in ${txId}#0
+        --tx-in-collateral ${txId}#0
+        --tx-out ${addr1}+50 + 95 abcd1234.testToken
+        --tx-out ${addr2}+1000 + 5 abcd1234.testToken
+        --change-address ${addr1}
+        --required-signer signing-keys/signing-key-${pkh1'}.skey
+        --mainnet --protocol-params-file ./protocol.json --out-file tx.raw
+      |]
+
+mintTokens :: Assertion
+mintTokens = do
+  let txOutRef = TxOutRef "e406b0cf676fc2b1a9edb0617f259ad025c20ea6f0333820aa7cef1bfe7302e5" 0
+      txOut = TxOut (Ledger.pubKeyHashAddress pkh1) (Ada.lovelaceValueOf 1250) Nothing
+      initState = def {utxos = [(txOutRef, txOut)]}
+      txId = encodeByteString $ fromBuiltin $ getTxId $ txOutRefId txOutRef
+
+      mintingPolicy :: Scripts.MintingPolicy
+      mintingPolicy =
+        Scripts.mkMintingPolicyScript
+          $$(PlutusTx.compile [||(\_ _ -> ())||])
+
+      curSymbol :: Value.CurrencySymbol
+      curSymbol = Ledger.scriptCurrencySymbol mintingPolicy
+
+      curSymbol' :: Text
+      curSymbol' = encodeByteString $ fromBuiltin $ Value.unCurrencySymbol curSymbol
+
+      redeemerHash =
+        let (Scripts.RedeemerHash rh) = Ledger.redeemerHash Ledger.unitRedeemer
+         in encodeByteString $ fromBuiltin rh
+
+      contract :: Contract () (Endpoint "SendAda" ()) Text ()
+      contract = do
+        let lookups =
+              Constraints.mintingPolicy mintingPolicy
+        let constraints =
+              Constraints.mustMintValue (Value.singleton curSymbol "testToken" 5)
+                <> Constraints.mustPayToPubKey
+                  pkh2
+                  (Ada.lovelaceValueOf 1000 <> Value.singleton curSymbol "testToken" 5)
+        void $ submitTxConstraintsWith @Void lookups constraints
+
+      (result, state, _) = runContractPure contract initState
+
+  result @?= Right ()
+  (state.commandHistory !! 1)
+    @?= Text.replace
+      "\n"
+      " "
+      [text| cardano-cli transaction build --alonzo-era
+       --tx-in ${txId}#0
+       --tx-in-collateral ${txId}#0
+       --tx-out ${addr2}+1000 + 5 ${curSymbol'}.testToken
+       --mint-script-file result-scripts/policy-${curSymbol'}.plutus
+       --mint-redeemer-file result-scripts/redeemer-${redeemerHash}.json
+       --mint 5 ${curSymbol'}.testToken
+       --change-address ${addr1}
+       --required-signer signing-keys/signing-key-${pkh1'}.skey
+       --mainnet --protocol-params-file ./protocol.json --out-file tx.raw
+      |]
+
+  assertFiles
+    state
+    [ [text|result-scripts/policy-${curSymbol'}.plutus|]
```

Comment:
Can we assert the content of these files or is it not worth it?

Reply:
Interesting, I have a feeling that would take more time than it's worth, but definitely a good idea.
Maybe in a later iteration (if there is one)